listener.go

```diff
@@ -74,6 +74,25 @@ func (l *listener) Accept() (tpt.CapableConn, error) {
 			sess.CloseWithError(errorCodeConnectionGating, "connection gated")
 			continue
 		}
+
+		// return through active hole punching if any
+		key := sess.RemoteAddr().String()
```

Comment:
Should be `string(sess.RemoteAddr().Bytes())` (no need to convert to a _real_ string, we just need a stable key).

Reply:
There's no `Bytes()` method on a `net.Addr`. Also, Go has multiple ways of represent an IPv4 address, so this wouldn't be a stable key..

Reply:
Oh, I see. I thought this was a multiaddr.

Reply:
Hm. Actually... we _need_ to include the peer ID here.

Reply:
Done.